int32 prox1
int32 dist1
int32 prox2
int32 dist2
int32 prox3
int32 dist3
